pkgs/development/compilers/gcc/12/default.nix

```diff
@@ -288,7 +288,10 @@ lib.pipe (stdenv.mkDerivation ({
 
   configurePlatforms = [ "build" "host" "target" ];
 
-  configureFlags = callFile ../common/configure-flags.nix { };
+  configureFlags = callFile ../common/configure-flags.nix { }
+    # On mips64-unknown-linux-gnu libsanitizer defines collide with
+    # glibc's definitions and fail the build. It was fixed in gcc-13+.
+    ++ lib.optionals targetPlatform.isMips [ "--disable-libsanitizer" ];
```

Comment:
We could be more specific here (like in my PR), since it's only an issue when the ABI isn't otherwise specified.  Not sure if it's worth it or not.  Up to @amjoseph-nixpkgs as an actual MIPS user, I'd say.

Reply:
Yeah, narrowing it down to `n32` ABI sounds reasonable. Would be nice if our `isMips64n32` helper worked as is. Not sure if it's broken helper or I'm using it incorrectly:

```
$ nix repl . --argstr crossSystem mips64-linux

nix-repl> stdenv.targetPlatform.config
"mips64-unknown-linux-gnu"

nix-repl> stdenv.hostPlatform.isMips64n32
false
```

Reply:
And `pkgs/development/compilers/gcc/common/platform-flags.nix` already contains the equivalent code:

```nix
(lib.optional targetPlatform.isMips64n32 "--disable-libsanitizer") # libsanitizer does not compile on mips64n32
```

That explains why `pkgsCross.mips64-linux-gnuabin32` is able to compile. Maybe we should just fix the predicate here.

Reply:
Fixed `mips64-linux` to return `true` for `isMips64n32`.

Reply:
https://github.com/NixOS/nixpkgs/commit/6cec03027a07162b1fe1d5bfd61c98e98eb5c316 looked like the right solution; I'm testing that right now...